### PR TITLE
[buteo-sync-plugins-social] Implemented OneDrive images sync plugin. …

### PR DESCRIPTION
**Pull request by antseppa**
…Fixes MER#1332

A few things still to do:

  - OneDrive should return list of albums with GET /me/albums. Currently it just seems to return the default one but not others I have created. Need to figure out what's missing.

- This handles album content count incorrectly. The returned count value contains all the files and folders in the album, not just images and UI ends up showing wrong number. Need to modify it show that we update the count after album contents is queried and we know the real count.

### CODE REVIEW

**Comment by antseppa** [comment]
Updated: delays adding albums to database until we know the actual image count in them.

**Comment by antseppa** [comment]
Updated: earlier version compared both updateTime and imageCount to decide if an album needs to be updated. We can't use image count as the value returned by the server is not limited to photos. Fixed.

File: rpm/sociald.spec [hunk 1]
```diff
@@ -344,6 +344,32 @@ systemctl-user try-restart msyncd.service || :
...
+#out-of-process-plugin form:
+/usr/lib/buteo-plugins-qt5/oopp/onedrive-images-client
+#in-process-plugin form:
+#/usr/lib/buteo-plugins-qt5/libonedrive-images-client.so
+%config %{_sysconfdir}/buteo/profiles/client/onedrive-images.xml
+%config %{_sysconfdir}/buteo/profiles/sync/onedrive.Images.xml
+
+%pre onedrive-images
+rm -f /home/nemo/.cache/msyncd/sync/client/onedrive-images.xml
+rm -f /home/nemo/.cache/msyncd/sync/onedrive.Images.xml
```
Comment (saukko) [inline rpm/sociald.spec]: Do not hardcode the /home/nemo/ path here. Do we really need this?

**Comment by antseppa** [comment]
Updated: removed "fields" parameter from user information query (it actually caused server to not return the fields we want).

**Comment by antseppa** [comment]
Updated: store image description field and user accoundId field.

**Comment by chriadam** [comment]
Closing as I've manually cherry picked this to master from the cloud branch, after squashing some commits together.

**Pull request closed without merging**